src/crypto/tls/handshake_server_tls13.go

```diff
@@ -279,23 +280,38 @@ GroupSelection:
 		clientKeyShare = &hs.clientHello.keyShares[0]
 	}
 
-	if _, ok := curveForCurveID(selectedGroup); selectedGroup != X25519 && !ok {
+	if _, ok := curveForCurveID(selectedGroup); selectedGroup != X25519 && curveIdToCirclScheme(selectedGroup) == nil && !ok {
```

Comment:
For consistency with the rest of the PR, why not do the CIRCL KEM check first, then the standard DH logic:
```go
if kem := curveIdtoCirclScheme(selectedGroup); kem != nil {
    // do CIRCL
} else if _, ok := curveForCurveID(selectedGroup); selectedGroup != X25519 && !ok {
   // do standard thing
}
```

Reply:
I find the current diff easier to read